internal/sci/aws/server_test.go

```diff
@@ -0,0 +1,147 @@
+	}
+	sess, err := session.NewSession()
+	assert.NoError(t, err)
+
+	s3Client := s3.New(sess)
+	server := &Server{
+		Clients: Clients{
+			S3Client: s3Client,
+		},
+	}
+
+	bucket := "substratus-test-bucket-" + randomString(8, charset)
+	object := "test-object"
+
+	_, err = server.Clients.S3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: &bucket,
+	})
+	assert.NoError(t, err)
+
+	defer server.Clients.S3Client.DeleteBucket(&s3.DeleteBucketInput{Bucket: &bucket})
+
+	// Upload an object
+	_, err = server.Clients.S3Client.PutObject(&s3.PutObjectInput{
+		Bucket: &bucket,
+		Key:    &object,
+		Body:   strings.NewReader("test-data"),
+	})
+	assert.NoError(t, err)
+
+	resp, err := server.GetObjectMd5(context.TODO(), &sci.GetObjectMd5Request{
+		BucketName: bucket,
+		ObjectName: object,
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, resp)
+	if resp != nil {
+		assert.NotEmpty(t, resp.Md5Checksum)
+	}
+}
+
+func TestBindIdentity(t *testing.T) {
+	envAccountID := os.Getenv("AWS_ACCOUNT_ID")
+	if envAccountID == "" {
+		t.Skip("Skipping TestBindIdentity because AWS_ACCOUNT_ID is not set")
+	}
+	envClusterName := os.Getenv("CLUSTER_NAME")
+	if envClusterName == "" {
+		t.Skip("Skipping TestBindIdentity because CLUSTER_NAME is not set")
+	}
+
+	sess, err := session.NewSession()
+	assert.NoError(t, err)
+	iamClient := iam.New(sess)
+
+	oidcProviderURL := "oidc.eks.us-west-2.amazonaws.com/id/C2A3CBF5FF8C55D72C8843756CD44444"
+	server := &Server{
+		Clients: Clients{
+			IamClient: iamClient,
+		},
+		OIDCProviderURL: oidcProviderURL,
+		OIDCProviderARN: "arn:aws:iam::243019462222:oidc-provider/" + oidcProviderURL,
+	}
+
+	roleName := "test-role" + randomString(8, charset)
+	rolePolicy := `{
+		"Version": "2012-10-17",
+		"Statement": [
+		  {
+			"Effect": "Allow",
+			"Principal": {
+			  "Service": "lambda.amazonaws.com"
+			},
+			"Action": "sts:AssumeRole"
+		  }
+		]
+	  }`
+
+	_, err = server.Clients.IamClient.CreateRole(&iam.CreateRoleInput{
+		RoleName:                 &roleName,
+		AssumeRolePolicyDocument: awssdk.String(rolePolicy),
+	})
+	assert.NoError(t, err)
+
+	defer func() {
+		if _, err := server.Clients.IamClient.DeleteRole(&iam.DeleteRoleInput{RoleName: &roleName}); err != nil {
+			t.Logf("Failed to delete IAM role: %v", err)
+		}
+	}()
+
+	// Debug: Fetch and print the current trust policy before making the BindIdentity call
+	getRoleInput := &iam.GetRoleInput{
+		RoleName: awssdk.String(roleName),
+	}
+	getRoleOutput, err := server.Clients.IamClient.GetRole(getRoleInput)
+	if err != nil {
+		t.Fatalf("Debug: failed to get the role: %v", err)
+	}
+	t.Logf("Debug: Current Trust Policy: %s", *getRoleOutput.Role.AssumeRolePolicyDocument)
+
+	resp, err := server.BindIdentity(context.TODO(), &sci.BindIdentityRequest{
+		Principal:                roleName,
+		KubernetesNamespace:      "test-namespace",
+		KubernetesServiceAccount: "test-serviceaccount",
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, resp)
```

Comment:
I saw how painful it is to decode the string into json so if you feel it's too much effort, feel free to ignore this comment.

There should be an assertion that verifies that after BindIdentity, the GetRole returns an AssumeRolePolicyDocument that includes `system:serviceaccount:test-namespace:test-serviceaccount` sub somewhere in the assumerole policy.

Reply:
Lazy way to accomplish: check that it contains a string in the stringified document

Reply:
Done